src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncController.scala

```diff
@@ -175,22 +176,38 @@ class FastSyncController(
     handshakedPeers -= peer
   }
 
-  class SyncingHandler(targetBlock: BlockHeader) {
+  class SyncingHandler(targetBlock: BlockHeader, syncStateStorage: FastSyncStateStorage) {
 
     private val blockHeadersHandlerName = "block-headers-request-handler"
 
-    private var nonMptNodesQueue: Set[HashType] = Set.empty
-    private var mptNodesQueue: Set[HashType] = Set.empty
+    private val initialSyncState: SyncState = if (continueAfterRestart)
+      syncStateStorage.getSyncState().getOrElse(SyncState.empty)
+    else
+      SyncState.empty
 
-    private var blockBodiesQueue: Set[ByteString] = Set.empty
-    private var receiptsQueue: Set[ByteString] = Set.empty
 
-    private var downloadedNodesCount: Int = 0
+    private var mptNodesQueue: Set[HashType] = initialSyncState.mptNodesQueue
+    private var nonMptNodesQueue: Set[HashType] = initialSyncState.nonMptNodesQueue
+    private var blockBodiesQueue: Set[ByteString] = initialSyncState.blockBodiesQueue
+    private var receiptsQueue: Set[ByteString] = initialSyncState.receiptsQueue
+    private var downloadedNodesCount: Int = initialSyncState.downloadedNodesCount
 
     private var assignedHandlers: Map[ActorRef, ActorRef] = Map.empty
-
     private var bestBlockHeaderNumber: BigInt = appStateStorage.getBestBlockNumber()
 
+    val hashTypeSetPersist: Option[Cancellable] = if (continueAfterRestart) {
```

Comment:
the name of this val seems no longer valid, how about `syncStatePersistCancellable`?